IntegrationTests/tests_01_allocation_counters/test_01_resources/test_stream_teardown_100_concurrent.swift

```diff
@@ -158,29 +156,71 @@ fileprivate class SendGoawayHandler: ChannelInboundHandler {
     )
 
     private let expectedStreams: Int
-    private var seenStreams: Int
+    private let seenStreams = ManagedAtomic<Int>(0)
```

Comment:
Why did this become an atomic?

Reply:
I did this so that you'd have more constant overhead from the test harness counting seen streams when comparing the two stream multiplexing methods. I'm happy to change this back if you'd like.

Reply:
I guess I don't follow why this change would produce a more-constant overhead?

Reply:
The intent was so that both versions of this test would use atomics for this counter. However after discussions neither now do. https://github.com/apple/swift-nio-http2/pull/384 removed the need for the `StreamDelegate` to be `Sendable` so I have removed these atomics.